scheduler-server/scheduling-dsl-compiler/src/main.ts

```diff
@@ -1,38 +1,61 @@
-import executeSourceCode from "./executor.js";
-import fs from "fs";
+import fs from 'fs';
+import ts from 'typescript';
+import { UserCodeRunner } from '@nasa-jpl/aerie-ts-user-code-runner';
+import type { Goal } from './libs/scheduler-edsl-fluent-api.js'
+
+const codeRunner = new UserCodeRunner();
+const schedulerEDSL = fs.readFileSync(`${process.env.SCHEDULING_DSL_COMPILER_ROOT}/src/libs/scheduler-edsl-fluent-api.ts`, 'utf8');
+const schedulerAST = fs.readFileSync(`${process.env.SCHEDULING_DSL_COMPILER_ROOT}/src/libs/scheduler-ast.ts`, 'utf8');
 
 process.on('uncaughtException', (err) => {
-  console.error("uncaughtException")
+  console.error('uncaughtException');
   console.error((err && err.stack) ? err.stack : err);
-  process.stdout.write("panic\n" + err.stack);
-  process.exit(1)
+  process.stdout.write('panic\n' + err.stack ?? err.message);
+  process.exit(1);
 });
 
 async function handleRequest(data: Buffer) {
   try {
-    const { source, filename, generatedCode } = JSON.parse(data.toString()) as { source: string, filename: string, generatedCode: string };
-    const result = await executeSourceCode({
-      source,
-      filename,
-      generatedCode,
-      dslTS: (await fs.promises.readFile(`${process.env.SCHEDULING_DSL_COMPILER_ROOT}/src/libs/scheduler-edsl-fluent-api.ts`)).toString()
-    })
-    const stringified = JSON.stringify(result);
+    const { goalCode, missionModelGeneratedCode } = JSON.parse(data.toString()) as { goalCode: string, missionModelGeneratedCode: string };
+
+    const result = await codeRunner.executeUserCode<[], Goal>(
+        goalCode,
+        [],
+        'Goal',
+        [],
+        10000,
+        [
+          ts.createSourceFile('scheduler-ast.ts', schedulerAST, ts.ScriptTarget.ESNext),
+          ts.createSourceFile('scheduler-edsl-fluent-api.ts', schedulerEDSL, ts.ScriptTarget.ESNext),
+          ts.createSourceFile('mission-model-generated-code.ts', missionModelGeneratedCode, ts.ScriptTarget.ESNext),
+        ],
+    );
+
+    if (result.isErr()) {
+      process.stdout.write('error\n' + JSON.stringify(result.unwrapErr().map(err => err.toJSON())) + '\n');
+      return;
```

Comment:
Aha. Here's the bug. We need to re-register the `data` callback here. Otherwise, once we've returned an error, we're no longer listening to standard input.

Reply:
Good catch! And debugging!